### PR TITLE
Obj locator - new interfaces, up to date with mainline

### PR DESCRIPTION
**Pull request by tim0c**
- Interface to support communication with object storage.
- Mongo logic 
- Constructor dependencies (WIP)
- NEED mongo database and collection name from Start-Up/Registrar

### CODE REVIEW

File: core/src/edu/rice/rbox/Location/locator/Tests/LocatorMainImplTests.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+//
+//import com.mongodb.client.model.Filters;
+//import edu.rice.rbox.Location.locator.LocatorMainImpl;
+//import org.junit.jupiter.api.Test;
+//
+//import java.util.ArrayList;
+//import java.util.HashMap;
+//import java.util.UUID;
+//
+//public class LocatorMainImplTests {
```
Comment (mja9) [inline core/src/edu/rice/rbox/Location/locator/Tests/LocatorMainImplTests.java:11]: I'd prefer if we kept commented out files out of main but it's not necessary

Comment (mja9) [inline core/src/edu/rice/rbox/Location/locator/Tests/LocatorMainImplTests.java]: If possible, can you exclude commented out files from the main branch?

File: core/src/main/proto/Locator.proto [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+message QueryResponse {
+  // Mapping of results to a string
+  // Map<GameObjectUUID, ServerObjectUUID>
+  map<string, string> result = 1;
+  BasicInfo info = 2;
+}
+
+message Empty {
```
Comment (mja9) [inline core/src/main/proto/Locator.proto:43]: You should replace this empty message with Google's defined empty message: 
`import "google/protobuf/empty.proto";`
I had problems implementing the Empty message

File: core/src/main/proto/Locator.proto [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  string confirmation = 1;
+}
+
+message QueryRegistrar {
+  // BSON Query String
+  string query = 1;
+  BasicInfo info = 2;
+}
+
+message QueryResponse {
```
Comment (popatr15) [inline core/src/main/proto/Locator.proto:36]: I dont think we need any of these messages any more. Instead you should add any necessary messages to the main proto file

**Review by popatr15 — DISMISSED** [review]

**Review by willsu88 — DISMISSED** [review]

**Review by popatr15 — APPROVED** [review]

**Pull request merged**